mesonbuild/modules/gnome.py

```diff
@@ -1055,47 +1056,92 @@ def yelp(self, state: 'ModuleState', args: T.Tuple[str, T.List[str]], kwargs: 'Y
                 raise MesonException('Yelp requires a list of sources')
         elif args[1]:
             mlog.warning('"gnome.yelp" ignores positional sources arguments when the "sources" keyword argument is set')
-        source_str = '@@'.join(sources)
+        sources_files = [mesonlib.File.from_source_file(state.environment.source_dir,
+                                                        os.path.join(state.subdir, 'C'),
+                                                        s) for s in sources]
 
         langs = kwargs['languages']
+        if not langs:
+            langs = read_linguas(state.subdir)
+
+        media = kwargs['media']
+        symlinks = kwargs['symlink_media']
+        targets: T.List['Target'] = []
+        potargets: T.List['build.RunTarget'] = []
+
+        install_dir = kwargs.get('install_dir', os.path.join(state.environment.get_datadir(), 'help'))
```

Comment:
This isn't going to do what you expect, typed_kwargs ensures that there's always a value. You'll need kwargs or fallback

Reply:
Thank you for this comment. Originally this module hard-coded the install_dir to `os.path.join(state.environment.get_datadir(), 'help')`. I do not know why did I change the behavior without documenting it and I cannot see any reason for kwargs. So I have reverted to the original behavior.